lib/logstash/plugin_mixins/validator_support/host_port_pair_validation_adapter.rb

```diff
@@ -0,0 +1,22 @@
+# encoding: utf-8
+
+require 'logstash/plugin_mixins/validator_support'
+
+module LogStash
+  module PluginMixins
+    module ValidatorSupport
+
+      host_port_regex = %r{(?:(?:[a-z][a-z0-9\-\.])|(([0-9a-fA-F]{0,4}:){1,7}[0-9a-fA-F]{0,4})|(?:\b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b))(:?:[0-9]{1,5})?$}xi
```

Comment:
Review note: used online regex tool for more tests: https://regex101.com/r/c0Vrac/1